File: backend/src/main/java/com/mlicer/uoc/lavurgerapi/dto/OrderDTO.java
package com.mlicer.uoc.lavurgerapi.dto;

import jakarta.validation.constraints.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

public record OrderDTO(
        Long id,

        @NotBlank(message = "Order number is required")
        String orderNumber,

        String status,

        @NotBlank(message = "Order type is required")
        String orderType,

        String paymentMethod,
        String paymentStatus,

        @NotNull(message = "Total amount is required")
        @DecimalMin(value = "0.0", inclusive = true, message = "Total amount cannot be negative")
        @Digits(integer = 8, fraction = 2)
        BigDecimal totalAmount,

        String customerComment,
        Long customerId,
        Long tableId,
        List<OrderItemDTO> items,
        LocalDateTime createdAt
) {}

File: frontend/src/app/shared/models/dtos/order.dto.ts
import { OrderType } from '../enums/order-type.enum';
import { OrderStatus } from '../enums/order-status.enum';
import { PaymentMethod } from '../enums/payment-method.enum';
import { PaymentStatus } from '../enums/payment-status.enum';
import { ProductDTO } from './product.dto';

export interface OrderItemDTO {
  id?: number;
  productId: number;
  product?: ProductDTO;
  quantity: number;
  unitPrice: number;
  subtotal: number;
  notes?: string;
}

export interface OrderDTO {
  id: number;
  orderType: OrderType;
  status: OrderStatus;
  paymentMethod: PaymentMethod;
  paymentStatus: PaymentStatus;
  totalAmount: number;
  transactionId?: string;
  customerEmail?: string;
  customerComment?: string;
  createdAt: string;
  updatedAt: string;

  items: OrderItemDTO[];
  tableId?: number | null;
}
